### PR TITLE
Add fresh releases backend

### PR DESCRIPTION
**Pull request by mayhem**
This PR does the following:

- Add support for upcoming releases (sitewide) API call.
- Add support for upcoming release view.
- Create a new explore and explore API blueprint and move the huesound and similar users pages to it.

### CODE REVIEW

**Comment by chinmaykunkikar** [comment]
Question about nomenclature: Are we calling this "upcoming releases" or "recent releases" as mentioned in #2037?

File: listenbrainz/webserver/views/metadata_api.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import datetime
```
Comment (chinmaykunkikar) [inline listenbrainz/webserver/views/metadata_api.py:1]: Extraneous import?

Comment (mayhem) [reply listenbrainz/webserver/views/metadata_api.py]: Fixed.

**Comment by mayhem** [comment]
chinmay: Very good question -- after a bit of an IRC conversation we've decided on "Fresh Releases", which encompasses both and is a bit more "hip". I'll update this PR accordingly.

File: listenbrainz/db/fresh_releases.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                                     , artist_credit_name
+                                     , release_group_primary_type
+                                     , release_group_secondary_type
+                        ) AS q
+                  WHERE q.rnum = 1
+               ORDER BY release_date
+                      , artist_credit_name
+                      , release_name"""
+
+    psycopg2.extras.register_uuid()
```
Comment (amCap1712) [inline listenbrainz/db/fresh_releases.py:89]: Probably register somewhere in app or database engine creation?

**Review by amCap1712 — APPROVED** [review]
Adding some tests would be good otherwise looks good.

**Comment by mayhem** [comment]
Oh, whoops. text_explore.py didn't get commited. Will check in on monday when I get back to the office.

File: listenbrainz/webserver/templates/explore/fresh-releases.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{%- extends 'base-react.html' -%}
+
+{% block content %}
```
[the remark this answers is not in the document]
Comment (mayhem) [reply listenbrainz/webserver/templates/explore/fresh-releases.html:3]: Ah, this whole page was mostly just a placeholder. :)

File: listenbrainz/db/fresh_releases.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    with psycopg2.connect(current_app.config["MB_DATABASE_URI"]) as conn:
+        with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as curs:
+            curs.execute(query, (from_date, to_date))
+            row = curs.fetchone()
```
[the remark this answers is not in the document]
Comment (mayhem) [reply listenbrainz/db/fresh_releases.py:93]: Yep.

File: listenbrainz/db/fresh_releases.py (hunk 2 above)
Comment (mayhem) [reply listenbrainz/db/fresh_releases.py]: Done.

**Pull request merged**